### PR TITLE
Logging level

### PR DESCRIPTION
**Pull request by rsmekala**
Fix for issue #354 
Added level option to logging specifications. User can use `level` option in task to set the verbosity level for Juniper.junos role.

User can choose of either INFO/DEBUG logging level. The default level would be WARNING.

The order of preference for setting the level would be:
1. Set level to DEBUG if ANSIBLE_DEBUG is True.
2. Set level based on number of v's passed to module.
3. Set level based on level option mentioned in task data.

### CODE REVIEW

File: module_utils/juniper_junos_common.py [hunk 1]
```diff
@@ -497,12 +511,15 @@ class ModuleDocFragment(object):
...
-    'logdir': dict(type='path', required=False, default=None)
+    'logdir': dict(type='path', required=False, default=None),
+    'level': dict(choices=[None, 'INFO', 'DEBUG'], required=False, default=None)
 }
 
 # The logdir and logfile options are mutually exclusive.
 logging_spec_mutually_exclusive = ['logfile', 'logdir']
 
+
+
```
Comment (stacywsmith) [inline module_utils/juniper_junos_common.py]: Why add extra unnecessary whitespace? Suggest you always run `pep8` against your changes.

File: module_utils/juniper_junos_common.py [hunk 2]
```diff
@@ -652,7 +669,7 @@ def __init__(self,
         # Update argument_spec with the top_spec
         argument_spec.update(top_spec)
         # Extend mutually_exclusive with connection_mutually_exclusive
-        mutually_exclusive += top_spec_mutually_exclusive
+        mutually_exclusive = top_spec_mutually_exclusive
```
Comment (stacywsmith) [inline module_utils/juniper_junos_common.py]: This will break things! A module may specify parameters which are mutually exclusive. The common parameters which are mutually exclusive should be **added** to the module specified parameters, not override them.

**Comment by vnitinv** [comment]
@rsmekala Can you please clean up unwanted commits

**Comment by stacywsmith** [comment]
@rsmekala you added a commit that reverts the `supports_check_mode` fix, rather than updating your sandbox to include this commit. Now, if you were to merge this PR as it currently stands, you would actually revert the previous fix.

**Comment by vnitinv** [comment]
Fix #359

**Comment by rsmekala** [comment]
@stacywsmith The pull request adding supports_check_module to the role is still not merged (#356 )

We are planning to go with #357 to address the issue #340.

**Comment by vnitinv** [comment]
@rsmekala This pull request is clean as of now with respect to other changes in the discussion. So I think we are good.
